### PR TITLE
test the Collections > Approval page

### PR DESCRIPTION
**Pull request by ShaiahWren**
No-Issue

Fills in a few more test from https://github.com/himdel/ansible-hub-ui/issues/1, Collections > Approval page.

- [ ] has a default Needs Review filter
- [ ] links to the import log page
- [ ] updates certification status - reject and approve labels

### CODE REVIEW

**Comment by ShaiahWren** [comment]
@himdel, this passes locally. It looks like the 'Sign and approve' button isn't visible for gh and I don't know why. Any insights?

**Comment by himdel** [comment]
sure, you're running with signing enabled, the CI isn't :)

So.. what do you actually want to test? If this is about collection approval, you don't need signing, so this should assume signing is not enabled.

(If you do need signing, we'll have to use `cy.settings` to ensure signing is enabled.)

**Comment by brumik** [comment]
@ShaiahWren I think you can test for only "Approve" (does not need any flags), since "Sign and Approve" requires at least 3 extra flags (I am debugging it) and we need tests for only "Approve" too. I am testing the "Sing and Approve" in the signing cypress tests here: #2069.

**Comment by MilanPospisil** [comment]
Ah, this looks like exactly what Im already doing in this PR: https://github.com/ansible/ansible-hub-ui/pull/1923. It is nearly done, and I can see, yours too :-)

File: test/cypress/integration/collection_approval.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+describe('tests the approval list screen ', () => {
+  before(() => {
+    cy.settings({
+      GALAXY_AUTO_SIGN_COLLECTIONS: true,
+    });
+  });
```
Comment (himdel) [inline test/cypress/integration/collection_approval.js:6]:
```suggestion

  before(() => {
    cy.settings({
      GALAXY_REQUIRE_CONTENT_APPROVAL: true,
    });
  });
  
```

File: test/cypress/integration/collection_approval.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  after(() => {
+    cy.settings();
+  });
+
+  it('has a default Needs Review filter', () => {
+    cy.get('.pf-c-chip-group__list').contains('Needs Review');
+  });
+
+  it.skip('rejects certification status and approves it again', () => {
```
Comment (himdel) [inline test/cypress/integration/collection_approval.js:22]:
```suggestion
  it('rejects certification status and approves it again', () => {
```

File: test/cypress/integration/collection_approval.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    //reject
+
+    cy.get('button[aria-label="Actions"]:first').click();
+    cy.contains('Reject').click();
+    cy.get('tr').eq(1).contains('Rejected');
+
+    //approve
+    cy.get('button[aria-label="Actions"]:first').click();
+    cy.contains('Approve').click();
+    cy.get('tr').eq(1).contains('Signed and approved');
```
Comment (himdel) [inline test/cypress/integration/collection_approval.js:40]:
```suggestion
    cy.get('tr').eq(1).contains('Approved');
```

**Review by himdel — COMMENTED** [review]
I think you want something like this?

Signing is not available in CI, so we should not be enabling it for these tests.
The tests themselves can work, you don't need to `skip` one.

File: test/cypress/integration/collection_approval.js (hunk 1 above)
Comment (himdel) [reply test/cypress/integration/collection_approval.js:6]: (unless you don't need approval enabled either? .. but if this is testing the approval page, it should probably test the normal process where the collection has to be approved first)

**Review by himdel — APPROVED** [review]

**Pull request merged**